### PR TITLE
JDK-8220222: specify clearly gradle project dependencies

### PR DESCRIPTION
**Pull request by christianheilmann**
In order to import the openjdk-jfx project correctly in the different IDEs, the dependencies should be clearly defined. At the moment the project is working in IDEA and Netbeans correctly.

Currently, Netbeans has an issue open with gradle and jigsaw (https://issues.apache.org/jira/browse/NETBEANS-2004 and kelemen/netbeans-gradle-project#417).

So the only IDE working 100% correctly is IDEA for importing the openjdk-jfx project.

https://bugs.openjdk.java.net/browse/JDK-8220222

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1847,12 +1847,24 @@ project(":base") {
 
     sourceSets {
         main
-        shims
-        test
+        shims{
```
Comment (nlisker) [inline build.gradle]: Missing space before `{`. Same for all subsequent places.

File: build.gradle [hunk 2]
```diff
@@ -2396,13 +2431,14 @@ project(":swing") {
         //shims // no test shims needed
         test
     }
-
+
```
Comment (kevinrushforth) [inline build.gradle]: Please revert this unwanted white-space change.

File: build.gradle [hunk 3]
```diff
@@ -2396,13 +2431,14 @@ project(":swing") {
...
     }
-
+            
     project.ext.moduleSourcePath = defaultModuleSourcePath
     project.ext.moduleSourcePathShim = defaultModuleSourcePathShim
 
     commonModuleSetup(project, [ 'base', 'graphics', 'swing' ])
 
     dependencies {
+        compile project(":media")
```
Comment (kevinrushforth) [inline build.gradle]: `javafx.swing` does not depend on `javafx.media`. I think this should instead be:

```
                compile project(":base")
                compile project(":graphics")
```

File: build.gradle [hunk 4]
```diff
@@ -3119,6 +3179,9 @@ project(":web") {
     commonModuleSetup(project, [ 'base', 'graphics', 'controls', 'media', 'web' ])
 
     dependencies {
+        compile project(":base")
+        compile project(":graphics")
+        compile project(":media")
```
Comment (kevinrushforth) [inline build.gradle]: Should a dependency on `controls` be added here as well?

**Comment by christianheilmann** [comment]
@kevinrushforth thanks for your review. You are right in all points and I integrated all your reviews and merged it with the develop branch of upstream.

This checkin enables full working on OpenJFX in IntelliJ IDEA Version 20.18.3 or higher.
Simply open the build.gradle in IntelliJ and you are ready to code.

The checked in IntelliJ project files should be removed, because they do not work and are misleading.
The remove of the IntelliJ project files should be done with https://github.com/javafxports/openjdk-jfx/issues/4

**Pull request merged**